pkg/common/custom_conditions.go

```diff
@@ -56,7 +57,7 @@ func SetTerminalState(
 	resourceName *string,
 	terminalStatus string,
 ) bool {
-	if latestStatus == nil || *latestStatus != terminalStatus {
+	if latestStatus == nil || strings.ReplaceAll(*latestStatus, " ", "") != terminalStatus {
```

Comment:
Oh nice unit tests caught a bug already

------------------------------------------------

why does`latestStatus` have a spaces in the first place ?

shouldn't `latestStatus` should already be set to one of these values `'Creating'|'Created'|'CreateFailed'|'Deleting'|'DeleteFailed'` ?

Reply:
This function was called from pkg/resource/feature_group/custom_update_conditions.go:

The latest status is retrieved by: r.ko.Status.FeatureGroupStatus.
When I printed this (for the CreateFailed state), it had a space in it ("Create Failed").
Whereas the terminal condition is retrieved by: svcsdk.FeatureGroupStatusCreateFailed (which has no spaces "CreateFailed")

Reply:
This was caused by a typo in unit testing and has since been corrected. (Status was listed as "Create Failed" with a space rather than the expected "CreateFailed")